sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -99,6 +105,68 @@
         }
     }
 
+    /**
+     * Asynchronously acquire a token from Active Directory with Aure CLI.
+     *
+     * @param request the details of the token request
+     * @return a Publisher that emits an AccessToken
+     */
+    public Mono<AccessToken> authenticateWithAzureCli(TokenRequestContext request) {
+        String azCommand = "az account get-access-token --output json --resource ";
+
+        StringBuilder command = new StringBuilder();
+        command.append(azCommand);
+        String scopes = ScopeUtil.scopesToResource(request.getScopes());
+        command.append(scopes);
+
+        AccessToken token = null;
+        try {
+            String starter;
+            String switcher;
+            if (System.getProperty("os.name").contains("Windows")) {
+                starter = "cmd.exe";
+                switcher = "/c";
+            } else {
+                starter = "/bin/sh";
+                switcher = "-c";
+            }
+
+            ProcessBuilder builder = new ProcessBuilder(starter, switcher, command.toString());
+            builder.redirectErrorStream(true);
+            Process p = builder.start();
+
+            BufferedReader r = new BufferedReader(new InputStreamReader(p.getInputStream()));
+            String line;
+            StringBuilder output = new StringBuilder();
+            while (true) {
+                line = r.readLine();
+                if (line == null) {
+                    break;
+                }
+                if (line.startsWith("'az' is not recognized") || line.matches("(.*)az:(.*)not found")) {
+                    throw new Exception("Azure CLI not installed");
+                }
+                output.append(line);
+            }
+            String processOutput = output.toString();
+            if (p.exitValue() != 0) {
+                throw new Exception(processOutput);
+            }
+            Map<String, String> objectMap = SERIALIZER_ADAPTER.deserialize(processOutput, Map.class,
+                    SerializerEncoding.JSON);
+            String accessToken = objectMap.get("accessToken");
+            String time = objectMap.get("expiresOn");
+            String timeToSecond = time.substring(0, time.indexOf("."));
+            String timeJoinedWithT = String.join("T", timeToSecond.split(" "));
+            OffsetDateTime expiresOn = LocalDateTime.parse(timeJoinedWithT, DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+                    .atZone(ZoneId.systemDefault()).toOffsetDateTime().withOffsetSameInstant(ZoneOffset.UTC);
+            token = new AccessToken(accessToken, expiresOn);
+        } catch (Exception e) {
+            return Mono.error(e);
```

Comment:
Okay, so, its being caught here.
better to add here, Mono.error(logger.logExceptionAsError(e)),
logger can be used from azure core